lib/index.js

```diff
@@ -2,42 +2,66 @@ import duplex from 'duplexer'
 import concat from 'concat-stream'
 import { PassThrough } from 'readable-stream'
 import Canvas from 'canvas'
-import brick from './brick'
 
-const getAverageColor = data => data
+const brick = `${_dirname}/../images/brick.png`
+const defaultOpts = { format: 'jpeg', quality : 'good', brick }
+const pixelIntervalPreset = { fast: 8, good: 5, best: 3, nearest: 2, bilinear: 1 }
+
+// Make a function to get the average color from given image data array
+// with given sample rate.
+const makeGetAverageColor = pixelInterval => data => data
+  .filter((_, index) => index % (pixelInterval * 4) < 4)
   .reduce((rgb, channel, index) => (rgb[index % 4] += channel, rgb), [0, 0, 0])
   .map(channel => Math.floor(channel / (data.length / 4)))
 
-const renderBrick = (ctx, x, y) => {
-  const color = getAverageColor(ctx.getImageData(x, y, brick.width, brick.height).data)
+// Make a function to render a brick at a point using given brick, average
+// color function and context.
+const makeRenderBrick = (brick, getAverageColor, ctx) => (x, y) => {
+  const { data } = ctx.getImageData(x, y, brick.width, brick.height)
+  const color = getAverageColor(data)
 
   ctx.clearRect(x, y, brick.width, brick.height)
   ctx.drawImage(brick, x, y)
-  ctx.fillStyle = `rgb(${color.join(',')})`
+  ctx.fillStyle = formatColor(color)
   ctx.fillRect(x, y, brick.width, brick.height)
 }
 
-const render = (format, body) => {
-  const image = Object.assign(new Canvas.Image(), { src: body })
-  const xCount = Math.round(image.width / brick.width)
-  const yCount = Math.round(image.height / brick.height)
+// Format given RGB array as a string.
+const formatColor = ([r, g, b]) => `rgb(${r},${g},${b})`
+
+// Make an `Image` object with given source.
+const img = src => Object.assign(new Canvas.Image(), { src })
+
+// Get the size in bricks of an image.
+const size = (image, brick) =>
+  [ image.width / brick.width, image.height / brick.height ].map(Math.floor)
+
+const render = (opts, src) => {
+  const image = img(src)
+  const brick = img(opts.brick)
+  const [ xCount, yCount ] = size(image, brick)
   const canvas = new Canvas(xCount * brick.width, yCount * brick.height)
   const ctx = canvas.getContext('2d')
+  const pixelInterval = opts.pixelInterval || pixelIntervalPreset[opts.quality]
```

Comment:
I’d argue we don’t need this variable and can pass its value directly in the function call next line. Thought?